### PR TITLE
Add types to graph viz code

### PR DESCRIPTION
**Pull request by OskarDamkjaer**

### CODE REVIEW

File: src/browser/modules/D3Visualization/lib/visualization/components/Node.ts [hunk 1]
```diff
@@ -46,6 +48,11 @@ export default class NodeVisualisationModel {
   minified: boolean
   contextMenu?: { menuSelection: string; menuContent: string; label: string }
 
+  // extra props TODO check if they make sense
```
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/Node.ts:51]: TODO

File: src/browser/modules/D3Visualization/lib/visualization/components/Node.ts [hunk 2]
```diff
@@ -73,15 +80,9 @@ export default class NodeVisualisationModel {
...
-        (relationship as any).source === this ||
-        (relationship as any).target === this
-      )
-        count++
-    }
-    return count
+  relationshipCount(graph: Graph): number {
+    return graph
+      .relationships()
+      .filter(rel => rel.source === this || rel.target === this).length
```
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/Node.ts:86]: 👍

File: src/browser/modules/D3Visualization/lib/visualization/components/collision.ts [hunk 3]
```diff
@@ -20,30 +20,48 @@
...
+  avoidOverlap: (nodes: d3.layout.force.Node[]): void => {
     const q = d3.geom.quadtree(nodes)
-    return Array.from(nodes).map(n => q.visit(collide(n)))
+    nodes.forEach(n => q.visit(collide(n)))
   }
 }
-const collide = (node: any) => {
+
+const collide = (node: d3.layout.force.Node) => {
+  // TODO solve/understand these errors
```
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/collision.ts:30]: TODO

File: src/browser/modules/D3Visualization/lib/visualization/components/collision.ts [hunk 4]
```diff
@@ -20,30 +20,48 @@
...
-  avoidOverlap: (nodes: any) => {
+  avoidOverlap: (nodes: d3.layout.force.Node[]): void => {
     const q = d3.geom.quadtree(nodes)
-    return Array.from(nodes).map(n => q.visit(collide(n)))
+    nodes.forEach(n => q.visit(collide(n)))
   }
 }
-const collide = (node: any) => {
+
+const collide = (node: d3.layout.force.Node) => {
```
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/collision.ts:29]: node type is Node & d3.layout.force.Node?

File: src/browser/modules/D3Visualization/lib/visualization/components/layout.ts [hunk 5]
```diff
@@ -21,101 +21,108 @@ import d3 from 'd3'
...
+        let maxStepsPerTick = 100
+        const maxAnimationFramesPerSecond = 60
+        const maxComputeTime = 1000 / maxAnimationFramesPerSecond
+        const now =
+          window.performance && window.performance.now
+            ? () => window.performance.now()
+            : () => Date.now()
+
+        const d3Tick = d3force.tick
+        // TODO fix this garb
```
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/layout.ts:75]: TODO

File: src/browser/modules/D3Visualization/lib/visualization/components/renderer.ts [hunk 6]
```diff
@@ -17,16 +17,27 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+import { VizObj } from './visualization'
+
+const noOp = () => undefined
+type RendererEventHandler = (selection: d3.Selection<any>, viz: VizObj) => void
+// TODO all should probably have a name
```
Comment (eijawerner) [inline src/browser/modules/D3Visualization/lib/visualization/components/renderer.ts:24]: TODO

**Review by eijawerner — COMMENTED** [review]
Nice to see so many types fixed! ⭐ There are a lot of TODO left, I started to mark them out, but finished after a while since they are so many, just do a global search on it :).

File: src/browser/modules/D3Visualization/lib/visualization/components/Node.ts (hunk 2 above)
Comment (OskarDamkjaer) [reply src/browser/modules/D3Visualization/lib/visualization/components/Node.ts:90]: If d3 changes how it does mutation, might want to compare ids instead here in the future

**Review by eijawerner — APPROVED** [review]
Nice work! Really great to get so many types in, and got rid of a LOT of warnings! 🥇 LGTM when the tests are green.

**Pull request merged**